PathKit/Sources/LineSegment.swift

```diff
@@ -27,13 +27,64 @@ public struct LineSegment {
         return lerp(start, end, at: t)
     }
 
-    /// Creates a line segment begins at the point `start` and ends the point `end`.
+    /// Creates a line segment beginning at `start` and ending at `end`.
     ///
     /// - Parameters:
     ///   - start: The starting point of the line segment.
     ///   - end: The end point of the line segment.
+    /// - Precondition: The `start` point and `end` point of the line segment must be different.
     public init(start: Point, end: Point) {
+        precondition(start != end)
```

Comment:
Why?

Reply:
It avoids division by zero in a couple of places.